src/HIE/Bios/Ghc/Api.hs

```diff
@@ -32,29 +31,25 @@ import HIE.Bios.Flags
 ----------------------------------------------------------------
 
 -- | Converting the 'Ghc' monad to the 'IO' monad. All exceptions are ignored and logged.
-withGHC :: FilePath  -- ^ A target file displayed in an error message.
+withGHC :: Cradle c  -- ^ The cradle to use for resolving the lib dir
```

Comment:
Not entirely sure I like this API. The ghc session is not dependent on the cradle, only on the libdir.
To me, this usage sends a signal that the cradle is somehow used in the session and needs to be known before launching the session. (my opinion is debatable here)
(btw, neither ghcide/hls nor haskell-ide-engine uses it)

Reply:
Ok, I'm happy enough to change this to the libdir, and then the users of this API can just run the cradle themselves to extract it

Reply:
Come to think of it, if we make the user pass in the libdir then there's no point really in having `withGHC`/`withGhcT`, since all it did was handle the libdir part

Reply:
I'd also advocate for just removing withGHC as well, since it's not used by ghcide/hls and is a carryover from ghc-mod

Reply:
Yeah, remove it.